### PR TITLE
[RCCA-10621] Stop converting resource IDs to numeric IDs for `kafka acl [create|delete|list]`

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Reopening #1678. No longer convert resource IDs to numeric IDs when creating ACLs, but continue to convert numeric IDs to resource IDs when listing ACLs.

### CODE REVIEW

File: test/kafka_test.go [hunk 1]
```diff
@@ -87,16 +87,13 @@ func (s *CLITestSuite) TestKafka() {
 		{args: "kafka acl list --cluster lkc-acls", fixture: "kafka/acl/list-cloud.golden"},
 		{args: "kafka acl list --cluster lkc-acls -o json", fixture: "kafka/acl/list-json-cloud.golden"},
 		{args: "kafka acl list --cluster lkc-acls -o yaml", fixture: "kafka/acl/list-yaml-cloud.golden"},
-		{args: "kafka acl list --principal User:12345", fixture: "kafka/acl/err-numeric-id.golden", exitCode: 1},
```
Comment (MuweiHe) [inline test/kafka_test.go:90]: is `principal` no longer supported?

Comment (brianstrauch) [reply test/kafka_test.go:90]: It is, but principals containing numeric IDs are not.

Comment (MuweiHe) [reply test/kafka_test.go:90]: would be good to still have a few `principal` tests?

Comment (brianstrauch) [reply test/kafka_test.go:90]: We still do!

**Review by MuweiHe — APPROVED** [review]

**Pull request merged**